addons/hr_holidays/models/hr_leave.py

```diff
@@ -80,10 +80,12 @@ def default_get(self, fields_list):
 
         lt = self.env['hr.leave.type']
         if self.env.context.get('holiday_status_display_name', True) and 'holiday_status_id' in fields_list and not defaults.get('holiday_status_id'):
-            lt = self.env['hr.leave.type'].search(['|', ('requires_allocation', '=', 'no'), ('has_valid_allocation', '=', True)], limit=1, order='sequence')
+            domain = ['|', ('requires_allocation', '=', 'no'), ('has_valid_allocation', '=', True)]
+            if 'request_unit_hours' in defaults:
```

Comment:
Value could be falsy
```suggestion
            if defaults.get('request_unit_hours'):
```